java-checks/src/main/java/org/sonar/java/checks/LeastSpecificTypeCheck.java

```diff
@@ -64,25 +65,35 @@ public void visitNode(Tree tree) {
       return;
     }
 
+    boolean springInjectionAnnotated = isSpringInjectionAnnotated(methodSymbol.metadata());
     methodTree.parameters().stream()
       .map(VariableTree::symbol)
       .filter(p -> p.type().isClass() && !p.type().symbol().isEnum() && !p.type().is("java.lang.String"))
-      .forEach(this::handleParameter);
+      .forEach(p -> handleParameter(p, springInjectionAnnotated));
   }
 
   private static boolean isOverloaded(Symbol.MethodSymbol methodSymbol) {
     return ((Symbol.TypeSymbol) methodSymbol.owner()).lookupSymbols(methodSymbol.name()).size() > 1;
   }
 
-  private void handleParameter(Symbol parameter) {
+  private void handleParameter(Symbol parameter, boolean springInjectionAnnotated) {
     Type leastSpecificType = findLeastSpecificType(parameter);
     if (parameter.type() != leastSpecificType
-      && !leastSpecificType.is("java.lang.Object")) {
-      String suggestedType = leastSpecificType.fullyQualifiedName().replace('$', '.');
+      && !leastSpecificType.is("java.lang.Object")
+      && !(springInjectionAnnotated && parameter.type().is("java.util.Collection"))) {
+      String suggestedType = getSuggestedType(springInjectionAnnotated, leastSpecificType);
       reportIssue(parameter.declaration(), String.format("Use '%s' here; it is a more general type than '%s'.", suggestedType, parameter.type().name()));
     }
   }
 
+  private String getSuggestedType(boolean springInjectionAnnotated, Type leastSpecificType) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "getSuggestedType" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=1971&issues=AWJyTr3PoPUqETT6TLyl&open=AWJyTr3PoPUqETT6TLyl)

[ISSUE_KEY]: AWJyTr3PoPUqETT6TLyl